src/codeu/chat/server/Transaction.java

```diff
@@ -0,0 +1,161 @@
+package codeu.chat.server;
+
+import codeu.chat.common.User;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.Tokenizer;
+import codeu.chat.util.Time;
+import codeu.chat.server.Controller;
+
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.FileWriter;
+import java.io.File;
+
+import java.io.IOException;
+
+
+public final class Transaction {
+
+  private ArrayList<String> transactionsList = new ArrayList<String>();
```

Comment:
This is not thread safe with the current implementation.  For example, it's possible for the server to try to write to this object while it's in the flush() iterator loop.  I'm not 100% sure what the Java behavior is, but in general Collections like ArrayList will throw an exception if you try to insert while there's iterator. 

Alternatively, it's also possible for this ArrayList to insert an object while it's in the flush() block before the ArrayLlist reset operation, which will result in that command being lost forever.

This can be handled two ways.  1) locks, which is probably too heavy for what we're doing.  Or 2) Use a concurrent collection object.  Timeline.java already makes use of some concurrent friendly collections, I would look at that for examples.  For the case of transactions, since it's also FIFO, a queue should work just fine.